pkg/testsuite/transactions_framework.go

```diff
@@ -342,159 +329,107 @@ func (t *TransactionFramework) DefaultAddress() iotago.Address {
 	}
 }
 
-func WithStakingEndEpoch(endEpoch iotago.EpochIndex) options.Option[iotago.AccountOutput] {
-	return func(accountOutput *iotago.AccountOutput) {
-		staking := accountOutput.FeatureSet().Staking()
-		if staking == nil {
-			panic("cannot update staking end epoch on account without Staking feature")
-		}
-		staking.EndEpoch = endEpoch
+func WithStakingFeature(amount iotago.BaseToken, fixedCost iotago.Mana, startEpoch iotago.EpochIndex, optEndEpoch ...iotago.EpochIndex) options.Option[builder.AccountOutputBuilder] {
+	return func(accountBuilder *builder.AccountOutputBuilder) {
+		accountBuilder.Staking(amount, fixedCost, startEpoch, optEndEpoch...)
 	}
 }
 
 // Account options
 
-func WithAccountMana(mana iotago.Mana) options.Option[iotago.AccountOutput] {
-	return func(accountOutput *iotago.AccountOutput) {
-		accountOutput.Mana = mana
-	}
-}
-
-func WithAccountAmount(amount iotago.BaseToken) options.Option[iotago.AccountOutput] {
-	return func(accountOutput *iotago.AccountOutput) {
-		accountOutput.Amount = amount
-	}
-}
-
-func WithAccountIncreasedStateIndex() options.Option[iotago.AccountOutput] {
-	return func(accountOutput *iotago.AccountOutput) {
-		accountOutput.StateIndex++
+func WithAccountMana(mana iotago.Mana) options.Option[builder.AccountOutputBuilder] {
+	return func(accountBuilder *builder.AccountOutputBuilder) {
+		accountBuilder.Mana(mana)
 	}
 }
 
-func WithAccountIncreasedFoundryCounter(diff uint32) options.Option[iotago.AccountOutput] {
-	return func(accountOutput *iotago.AccountOutput) {
-		accountOutput.FoundryCounter += diff
+func WithAccountAmount(amount iotago.BaseToken) options.Option[builder.AccountOutputBuilder] {
+	return func(accountBuilder *builder.AccountOutputBuilder) {
+		accountBuilder.Amount(amount)
 	}
 }
 
-func WithAccountFeatures(features iotago.AccountOutputFeatures) options.Option[iotago.AccountOutput] {
-	return func(accountOutput *iotago.AccountOutput) {
-		accountOutput.Features = features
-	}
-}
-
-func WithAccountImmutableFeatures(features iotago.AccountOutputImmFeatures) options.Option[iotago.AccountOutput] {
-	return func(accountOutput *iotago.AccountOutput) {
-		accountOutput.ImmutableFeatures = features
-	}
-}
-
-func WithAccountConditions(conditions iotago.AccountOutputUnlockConditions) options.Option[iotago.AccountOutput] {
-	return func(accountOutput *iotago.AccountOutput) {
-		accountOutput.Conditions = conditions
+func WithAccountConditions(conditions iotago.AccountOutputUnlockConditions) options.Option[builder.AccountOutputBuilder] {
+	return func(accountBuilder *builder.AccountOutputBuilder) {
+		for _, condition := range conditions.MustSet() {
+			switch condition.Type() {
+			case iotago.UnlockConditionStateControllerAddress:
+				accountBuilder.StateController(condition.(*iotago.StateControllerAddressUnlockCondition).Address)
+			case iotago.UnlockConditionGovernorAddress:
+				accountBuilder.Governor(condition.(*iotago.GovernorAddressUnlockCondition).Address)
+			}
+		}
```

Comment:
just checking if WithAccountIncreasedFoundryCounter, WithAccountIncreasedFoundryCounter and WithAccountImmutableFeatures were skipped on purpose?

Reply:
I removed it since it's not used during refactoring. :stuck_out_tongue: 
Will add it back.